cli/src/commands/auth.rs

```diff
@@ -0,0 +1,45 @@
+use crate::commands::Command;
+use crate::config::CliConfig;
+use crate::errors::{ExitCode, Fallible};
+use crate::style::KEY;
+use crate::telemetry::Session;
+use crate::terminal::sensitive;
+use console::style;
+use log::{info, warn};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Auth {
+    /// Setup your auth stuff
+    /// Link the CLI to your Apollo account using your Personal API Key.
+    Setup(Setup),
+}
+
+#[derive(StructOpt)]
+pub struct Setup {}
+
+impl Command for Setup {
+    fn run(&self, session: &mut Session) -> Fallible<ExitCode> {
+        session.log_command("auth setup");
+
+        if session.config.api_key.is_some() {
+            warn!("Authentication already configured.");
+        }
+
+        info!("To link your CLI to your Apollo account go to {} and create a new Personal API Key. Once you've done that, copy the key and paste it into the prompt below.",
```

Comment:
How does this wrap on terminals? Should we truncate it to 80 lines and add a new line?

Reply:
So we were debating that. We assumed most folks will have wrapping setup on terminals and that trying to do "smart" wrapping is something _tougher_ which might be best left for another pr.

We can throw newlines in, or try to truncate, but I do think there'll be a usability issue with logging if we require every call site to manage making the log lines pretty for users. Might be worth looking into various packages which can auto-wrap nicely etc.?

Reply:
That's fair! Well reasoned and fine by me!